test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Order.cshtml

```diff
@@ -34,6 +34,34 @@
             <select asp-for="@Model.Products" asp-items="@((IEnumerable<SelectListItem>)ViewBag.Items)" multiple="@(@object)"></select>
         </div>
         <div>
+            @if (Model.ProductDetails.Count > 0)
+            {
+                <table>
+                    <tr>
+                        <th>
+                            <label asp-for="ProductDetails.FirstOrDefault().Number"></label>
+                        </th>
+                        <th>
+                            <label asp-for="ProductDetails.FirstOrDefault().ProductName"></label>
+                        </th>
+                    </tr>
+                    @for (var i = 0; i < Model.ProductDetails.Count; i++)
+                    {
+                    <tr>
+                        <td>
+                            <input asp-for="ProductDetails[i].Number" />
```

Comment:
Does this sufficiently fix the issue? Given that you're iterating over the list our `TagHelper`s only ever see the `int` property Number, not the actual List.